docs/book/src/tasks/using-kustomize.md

```diff
@@ -0,0 +1,154 @@
+
+```yaml
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ../../base
+  - workload-mhc.yaml
+```
+
+The content of the `workload-mhc.yaml` file would be the definition of a standard MHC:
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: MachineHealthCheck
+metadata:
+  name: md-0-mhc
+spec:
+  clusterName: test
+  # maxUnhealthy: 40%
+  nodeStartupTimeout: 10m
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/deployment-name: md-0
+  unhealthyConditions:
+  - type: Ready
+    status: Unknown
+    timeout: 300s
+  - type: Ready
+    status: "False"
+    timeout: 300s
+```
+
+Running `kustomize build .` with this configuration would append the MHC to the base 
+Cluster API manifest, thus creating the MHC at the same time as the workload cluster.
+
+## Configuring Kustomize for Cluster API
+
+Although `kustomize` can be used with Cluster API without any changes, modifying the transformer 
+configurations for `kustomize` can make it more effective with Cluster API. For example, changes to 
+the `nameReference` transformer in `kustomize` will enable `kustomize` to know about the references 
+between Cluster API objects in a manifest. See [here](https://github.com/kubernetes-sigs/kustomize/tree/master/examples/transformerconfigs) 
+for more information on transformer configurations.
+
+Add the following content to the `namereference.yaml` transformer configuration:
+
+```yaml
+- kind: Cluster
+  group: cluster.x-k8s.io
+  version: v1alpha3
+  fieldSpecs:
+  - path: spec/clusterName
+    kind: MachineDeployment
+  - path: spec/template/spec/clusterName
+    kind: MachineDeployment
+
+- kind: AWSCluster
+  group: infrastructure.cluster.x-k8s.io
+  version: v1alpha3
+  fieldSpecs:
+  - path: spec/infrastructureRef/name
+    kind: Cluster
+
+- kind: KubeadmControlPlane
+  group: controlplane.cluster.x-k8s.io
+  version: v1alpha3
+  fieldSpecs:
+  - path: spec/controlPlaneRef/name
+    kind: Cluster
+
+- kind: AWSMachine
+  group: infrastructure.cluster.x-k8s.io
+  version: v1alpha3
+  fieldSpecs:
+  - path: spec/infrastructureRef/name
+    kind: Machine
+
+- kind: KubeadmConfig
+  group: bootstrap.cluster.x-k8s.io
+  version: v1alpha3
+  fieldSpecs:
+  - path: spec/bootstrap/configRef/name
+    kind: Machine
+
+- kind: AWSMachineTemplate
+  group: infrastructure.cluster.x-k8s.io
+  version: v1alpha3
+  fieldSpecs:
+  - path: spec/template/spec/infrastructureRef/name
+    kind: MachineDeployment
+  - path: spec/infrastructureTemplate/name
+    kind: KubeadmControlPlane
+
+- kind: KubeadmConfigTemplate
+  group: bootstrap.cluster.x-k8s.io
+  version: v1alpha3
+  fieldSpecs:
+  - path: spec/template/spec/bootstrap/configRef/name
+    kind: MachineDeployment
+```
+
+This will enable the use of the `namePrefix` and `nameSuffix` directives in `kustomization.yaml` 
+and ensure that _all_ references to the names are properly updated.
```

Comment:
What about adding a quick example here. e.g. changing namePrefix for a Cluster+ InfraClusterObject

Reply:
I've changed the title of this section and added some introductory text, as well as an example `kustomization.yaml` that shows using a modified nameReference transformer. Let me know what you think. These changes are in 0f98ee3aa.